### PR TITLE
Remove netty.transport.epoll in azure-cosmos module-info

### PR DESCRIPTION
**Pull request by Netyyyy**
Due to `netty-transport-epoll` changing its module name in version 4.1.77.Final https://github.com/netty/netty/issues/12456,
remove it from `module-info.java` for better supporting multiple Spring Boot versions

### CODE REVIEW

**Comment by Netyyyy** [comment]
@kushagraThapar please take a look and make sure it doesn't cause any other problems, thanks!

**Comment by saragluna** [comment]
Some more context on this:

| Netty Version | Description| Automatic Module Name|
|--|--|--|
|<= 4.1.69| Netty epoll classes and native shipped in one library|io.netty.transport.epoll|
|<=4.1.76| Netty epoll classes and native shipped in two libraries| io.netty.transport.epoll & io.netty.transport.classes.epoll|
|> 4.1.77| Netty epoll classes and native shipped in two libraries |  io.netty.transport.epoll.\<native-identifier> & io.netty.transport.classes.epoll

**Comment by saragluna** [comment]
Hi @kushagraThapar, please help take a look at this PR, we're trying to upgrade the dependencies to Spring Boot 2.7.0 this month, and probably we need to release a version for Spring Data Cosmos with Spring Data Commons 2.7.0.

**Review by FabianMeiswinkel — CHANGES_REQUESTED** [review]
This change needs further discussion - blocking the PR to make sure there is a consensus before it gets merged.

**Comment by kushagraThapar** [comment]
@saragluna / @Netyyyy  - ack on the PR review, @xinlian12 has another PR for testing purposes here - https://github.com/Azure/azure-sdk-for-java/pull/29509 
However, since this is a breaking change, I am wondering how do we want to tackle this. @alzimmermsft @joshfree  thoughts?

**Pull request closed without merging**
